File: make-transparent.py
from PIL import Image

def remove_background(input_path, output_path, tolerance=50):
    img = Image.open(input_path).convert("RGBA")
    datas = img.getdata()
    
    new_data = []
    # Let's assume background is white or black.
    # We will check the top-left pixel to guess the background color.
    bg_color = datas[0] 
    
    for item in datas:
        # Check distance to bg_color
        if abs(item[0]-bg_color[0]) < tolerance and abs(item[1]-bg_color[1]) < tolerance and abs(item[2]-bg_color[2]) < tolerance:
            new_data.append((255, 255, 255, 0)) # transparent
        else:
            new_data.append(item)
            
    img.putdata(new_data)
    img.save(output_path, "PNG")

remove_background("public/uploads/PhoneSpot.jpeg", "public/uploads/PhoneSpot.png", 50)
print("Saved PhoneSpot.png")

File: make-transparent.js
const { Jimp } = require('jimp');

async function removeBackground() {
    try {
        const image = await Jimp.read('public/uploads/PhoneSpot.jpeg');
        // Let's assume top-left pixel is background color
        const bgColor = image.getPixelColor(0, 0);
        
        // tolerance for similarity
        const tolerance = 50;
        
        // rgba of bgColor
        const r1 = (bgColor >> 24) & 255;
        const g1 = (bgColor >> 16) & 255;
        const b1 = (bgColor >> 8) & 255;
        
        image.scan(0, 0, image.bitmap.width, image.bitmap.height, function(x, y, idx) {
            const r2 = this.bitmap.data[idx + 0];
            const g2 = this.bitmap.data[idx + 1];
            const b2 = this.bitmap.data[idx + 2];
            
            if (Math.abs(r1 - r2) <= tolerance && Math.abs(g1 - g2) <= tolerance && Math.abs(b1 - b2) <= tolerance) {
                this.bitmap.data[idx + 3] = 0; // alpha to 0
            }
        });
        
        await image.write('public/uploads/PhoneSpot-trans.png');
        console.log('Background removed!');
    } catch (e) {
        console.error(e);
    }
}

removeBackground();
